packages/crowdstrike/data_stream/alert/elasticsearch/ingest_pipeline/default.yml

```
@@ -1401,6 +1402,32 @@ processors:
       tag: set_host_os_name_from_alert_operating_system
       copy_from: crowdstrike.alert.operating_system
       ignore_empty_value: true
+  - script:
+      description: Dynamically set host.os.type values.
+      tag: script_map_host_os_type
+      lang: painless
+      params:
+        os_type:
+          - linux
+          - macos
+          - unix
+          - windows
+          - ios
+          - android
+      source: |
+        if (ctx.crowdstrike?.alert?.device?.platform_name != null) {
+          for (String os: params.os_type) {
+            if (ctx.crowdstrike.alert.device.platform_name.toLowerCase().contains(os)) {
+              ctx.host.os.put('type', os);
+            }
+          }
```

Is there a reason that we cannot do a single map look-up? If there is, the linear scan I guess is OK, but we should not be making a new string for each iteration of the os_type list.

```suggestion
          String platform_name = ctx.crowdstrike.alert.device.platform_name.toLowerCase();
          for (String os: params.os_type) {
            if (platform_name.contains(os)) {
              ctx.host.os.put('type', os);
            }
          }
```
(same below)

Can you explain why we cannot use a map look-up?

---

I'm not sure what values we'll get in the `crowdstrike.alert.device.platform_name` field, so to cover all possible cases, I'm converting them to lowercase and checking if they contain any expected values from [host.os.type](https://www.elastic.co/guide/en/ecs/current/ecs-os.html#field-os-type).

applying same at other places...

---

OK. This is unfortunate, but if we have to do it, we have to do it.